### PR TITLE
GeoJSON clustering test

### PR DESCRIPTION
**Pull request by mourner**
- Refs https://github.com/mapbox/mapbox-gl-js/pull/1931. 
- Native test is disabled until implemented. 
- Also needs style-spec update so that it doesn't error on `cluster` and `clusterRadius` properties. https://github.com/mapbox/mapbox-gl-style-spec/pull/400

### CODE REVIEW

**Comment by mourner** [comment]
@jfirebaugh ready for review/merge.

File: render-tests/geojson/clustered/style.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      "cluster": true,
+      "clusterRadius": 200
+    }
+  },
+  "glyphs": "local://glyphs/{fontstack}/{range}.pbf",
+  "layers": [{
+    "id": "cluster",
+    "type": "circle",
+    "source": "geojson",
+    "filter": ["all", ["==", "cluster", true]],
```
Comment (jfirebaugh) [inline render-tests/geojson/clustered/style.json]: Flatten the redundant `"all"` clause in all the filters.

File: render-tests/geojson/clustered/style.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      "circle-color": "rgba(0, 200, 0, 1)",
+      "circle-radius": 20
+    }
+  }, {
+    "id": "cluster_label",
+    "type": "symbol",
+    "source": "geojson",
+    "filter": ["all", ["==", "cluster", true]],
+    "layout": {
+      "text-field": "{numPointsH}",
```
Comment (jfirebaugh) [inline render-tests/geojson/clustered/style.json]: `numPointsH` is pretty cryptic. How about `abbreviated_point_count`? (Underscores for consistency with mapbox-streets property name conventions.)

Comment (mourner) [inline render-tests/geojson/clustered/style.json]: Sounds good. Maybe `abbrev_point_count` for a bit more brevity? Or `num_points_abbrev`

Comment (jfirebaugh) [inline render-tests/geojson/clustered/style.json]: Why not just use whole words?

Comment (mourner) [inline render-tests/geojson/clustered/style.json]: Probably out of habit making everything short. :+1: to full words. :)

**Pull request merged**